File: ReturnNumberOfOccurrencesOfEachUniqueWordInString.rb
# Given a string return the count of each unique word 

def uniqueWordCount(str)
  obj = {}
  str = str.gsub(/\W/, ' ').downcase.strip.split(/\s+/)
  str.each do |i|
    if !obj.key?(i)
      obj[i] = 0
    end
    obj[i] += 1
  end
  obj
end

File: ReturnNumberOfOccurrencesOfEachUniqueWordInString.js
// Given a string return the count of each unique word 

function uniqueWordCount(str) {
  let obj = {};
  str = str.replace(/\W/g, ' ').toLowerCase().trim().split(/\s+/g);
  for (let i = 0; i < str.length; i++) {
    if (!(str[i] in obj)) {
      obj[str[i]] = 0;
    }
    obj[str[i]] += 1;
  }
  return obj;
}